### PR TITLE
remove thrift config from config

### PR DESCRIPTION
**Pull request by ramaraochavali**
Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

### CODE REVIEW

File: mesh/v1alpha1/config.proto [hunk 1]
```diff
@@ -361,18 +361,8 @@ message MeshConfig {
   // Note: Deprecated, please refer to Cert-Manager or other cert provisioning solutions to sign DNS certificates. 
   repeated Certificate certificates = 47;
 
-  message ThriftConfig {
```
Comment (hzxuzhonghu) [inline mesh/v1alpha1/config.proto:364]: How about hide it

Comment (ramaraochavali) [reply mesh/v1alpha1/config.proto:364]: We have removed the code completely and do not have plans to support it. So not see any benefits of hiding. Do you see any benefits of hiding vs. removing?

Comment (hzxuzhonghu) [reply mesh/v1alpha1/config.proto:364]: Will it break compatability removing a field?

Comment (ramaraochavali) [reply mesh/v1alpha1/config.proto:364]: AFAIK. it should not

**Comment by ericvn** [comment]
Note that this change is causing a failure in the client-go update: https://prow.istio.io/job-history/gs/istio-prow/logs/update_client-go_dep_client-go_postsubmit

**Comment by ramaraochavali** [comment]
@ericvn sorry. Here is the PR https://github.com/istio/istio/pull/41711 to fix it

**Pull request merged**